librz/util/vector.c

```diff
@@ -226,6 +230,21 @@ RZ_API void *rz_vector_push_front(RzVector *vec, void *x) {
 	return rz_vector_insert(vec, 0, x);
 }
 
+RZ_API void rz_vector_swap(RzVector *vec, size_t index_a, size_t index_b) {
+	rz_return_if_fail(vec && index_a < vec->len && index_b < vec->len);
+#ifndef _MSC_VER
+	ut8 tmp[vec->elem_size];
+#else
+	// MSVC is a C++ compiler and doesn't support C99 VLAs
+	ut8 *tmp = _alloca(vec->elem_size);
+#endif
```

Comment:
I think it will be easier to just use `malloc` instead of dynamic stack here. A boolean return value will be necessary to indicate if the swap succeeded or failed because of a failed malloc.